pkg/ccl/changefeedccl/avro.go

```diff
@@ -281,8 +351,61 @@ func (r *avroSchemaRecord) rowFromNative(native interface{}) (sqlbase.EncDatumRo
 	for fieldName, avroDatum := range avroDatums {
 		fieldIdx := r.fieldIdxByName[fieldName]
 		field := r.Fields[fieldIdx]
-		row[r.colIdxByFieldIdx[fieldIdx]] = sqlbase.DatumToEncDatum(
-			field.typ, field.decodeFn(avroDatum))
+		decoded, err := field.decodeFn(avroDatum)
+		if err != nil {
+			return nil, err
+		}
+		row[r.colIdxByFieldIdx[fieldIdx]] = sqlbase.DatumToEncDatum(field.typ, decoded)
 	}
 	return row, nil
 }
+
+// decimalToRat converts one of our apd decimals to the format expected by the
+// avro library we use. If the column has a fixed scale (which is always true if
+// precision is set) this is roundtripable without information loss.
+//
+// TODO(dan): We really should just be controlling our own encoding destiny
+// here. Make that possible.
+func decimalToRat(dec apd.Decimal, scale int32) (big.Rat, error) {
+	if dec.Form != apd.Finite {
+		return big.Rat{}, errors.Errorf(`cannot convert %s form decimal: %s`, dec.Form, &dec)
+	}
+	if e := -dec.Exponent; e > scale {
+		return big.Rat{}, errors.Errorf(`%s has more than %d digits after the decimal`, &dec, scale)
```

Comment:
Is this check required in addition to the next one? That is, is there a test case that would trigger this first error but not the second one? Or is it just to have a clearer error message?